File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int len = nums.size();
        int rsum =0;
        int lsum =0;
        int index =-1;
        for(int i =0;i<len;i++){
            rsum +=nums[i];
        }

        for(int i =0;i<len;i++){
            rsum -= nums[i];
            if(lsum == rsum ){
                return i;
            }
            lsum+=nums[i];
        }
       return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int total =0;
        for(int i =0;i<nums.length;i++){
            total += nums[i];
        }
        // System.out.println(total);
        int leftsum =0;
        for(int i =0;i<nums.length;i++){
            int rightsum = total-leftsum-nums[i];
            if(leftsum == rightsum){
                return i;
            }
            leftsum += nums[i];
        }
        return -1;
    }
}